frontend/heir/mlir_emitter.py

```diff
@@ -379,6 +380,9 @@ def emit_assign(self, assign):
         return ""
     raise NotImplementedError(f"Unsupported IR Element: {assign}")
 
+  # TODO (#1162): this function seems to be unused and should probably be removed.
```

Comment:
Oh yeah, it looks like the different ir.Expr types are handled in the switch statement for emit_assign - @j2kun maybe this should be removed?